Src/AutoNSubstitute/AutoNSubstitute.csproj

```diff
@@ -64,6 +64,7 @@
   <ItemGroup>
     <Compile Include="AutoConfiguredNSubstituteCustomization.cs" />
     <Compile Include="AutoNSubstituteCustomization.cs" />
+    <Compile Include="AutoSubstituteCustomization.cs" />

```

Comment:
To keep the number of classes small, I'd prefer to change behavior of the existing `AutoNSubstituteCustomization` and the `AutoConfiguredNSubstituteCustomization` to support the new `SubstituteAttribute`. Their `Builder` inspection properties mayhave to be modified to return an `ISpecimenBuilder` instance of a different type - `SubstituteBuilder` or `SubstituteAttributeRelay` instead of the `NSubstituteBuilder`. The type of property itself would remain an `ISpecimenBuilder`. 

If you consider that change to be _backward compatible_, I would recommend 
- merging the new `AutoSubstituteCustomization` logic into the two existing `NSubstituteCustomization` classes and eliminating the new customization class before merging this PR;
- replacing the existing `NSubstituteBuilder` implementation with that of the new `SubstituteBuilder` and eliminating the new builder class before merging this PR;
- renaming the new `SubstituteXxx` classes to follow the existing `NSubstituteXxx` naming convention.

Alternatively, if returning a different `ISpecimenBuilder` from the `Builder` properties of the existing customization classes _breaks_ backward compatibility, I would recommend 
- renaming the new `AutoSubstituteCustomization` to simply `SubstituteCustomization` to emphasize its connection to the other new classes
- leaving existing and new classes separate.

In other words, it's largely a tradeoff between the number of public classes and backward compatibility with implications on long term evolution of the API. How would you like to proceed?